src/main/java/Duke.java

```diff
@@ -1,10 +1,258 @@
+     *
+     * @params tasks  Array of current stored tasks
+     * @params userInput  Original input by user
+     * @params taskCount  Current count of tasks stored
+     * */
+    public static void addEventTask(Task[] tasks, String userInput, int taskCount) {
+        String description;
+        String at;
+        String[] informationStrings = determineTaskInformation(userInput, COMMAND_EVENT, EVENT_IDENTIFIER);
+        description = informationStrings[0].trim();
+        at = informationStrings[1].trim();
+        tasks[taskCount] = new Event(description, at);
+    }
+
+    /**
+     * Print confirmation text when a new task is added
+     *
+     * @params tasks  Array of current stored tasks
+     * @params taskCount  Current count of tasks stored
+     */
+    public static void printTaskAddedConfirmation(Task[] tasks, int taskCount) {
+        int numberOfTasks = taskCount + 1;
+        printSeparator();
+        System.out.println(MESSAGE_TASK_ADDED_CONFIRM);
+        System.out.println(MESSAGE_DOUBLE_WHITESPACE + tasks[taskCount]);
+        System.out.println(MESSAGE_NOW_YOU_HAVE + numberOfTasks + MESSAGE_IN_THE_LIST);
+        printSeparator();
+    }
+
+    /**
+     * Print list of tasks when user requests
+     *
+     * @params tasks  Array of current stored tasks
+     */
+    public static void printTaskList(Task[] tasks) {
+        int taskNumber = 1;
+        printSeparator();
+        if (tasks.length == 0) {
+            System.out.println(MESSAGE_ERROR_TASK_UNAVAILABLE);
+            return;
+        }
+        System.out.println(MESSAGE_TASKS_IN_LIST);
+        for (Task task : tasks) {
+            System.out.println(" " + taskNumber + "." + task);
+            taskNumber++;
+        }
+        printSeparator();
+    }
+
+    /**
+     * Sets isDone of selected task to true
+     *
+     * @params tasks  Array of current stored tasks
+     * @params splitUserInput  Array of strings after original user input has been split by whitespace
+     * @params taskCount  Current count of tasks stored
+     */
+    public static void setTaskAsDone(Task[] tasks, String[] splitUserInput, int taskCount) {
+        //Determine index of task to be marked as done - this part can pass in broken array straight (To change)
+        if (splitUserInput.length == 1) {
+            System.out.println(MESSAGE_ERROR_INVALID_COMMAND);
+            printSeparator();
+            return;
+        }
+        int taskNumber = Integer.parseInt(splitUserInput[1]) - 1;
+        printSeparator();
+
+        //Check if taskNumber is out of bounds
+        if (taskNumber < 0 || taskNumber > taskCount - 1) {
+            System.out.println(MESSAGE_ERROR_INVALID_NUMBER);
+            printSeparator();
+            return;
+        }
+
+        //TaskNumber is valid
+        tasks[taskNumber].setAsDone();
+        System.out.println(MESSAGE_TASK_MARKED);
+        System.out.println("  [" + "\u2713" + "] " + tasks[taskNumber].description);
```

Comment:
Since you declared the icon as a constant already in the other class, should the icon here not be used as a constant a well?

Reply:
That is a good catch, I have missed it out, I will change that thank you